plugins/org.jboss.reddeer.common/src/org/jboss/reddeer/common/wait/TimePeriod.java

```diff
@@ -42,7 +44,20 @@ private TimePeriod(long seconds) {
 	 * @return duration of time period in seconds
 	 */
 	public long getSeconds() {
-		return seconds;
+		if (seconds == Long.MAX_VALUE || seconds == 0) {
+			return seconds;
+		}
+
+		if (FACTOR == 1) {
+			return seconds;
+		}
+
+		// do not exceed max value of Long
+		if (seconds >= Long.MAX_VALUE / FACTOR) {
+			return Long.MAX_VALUE;
+		} else {
+			return Math.round(seconds * FACTOR);
```

Comment:
Use this expression Math.round((double)seconds * factor) otherwise round method will return Integer.MAX_VALUE for each input greater then Integer.MAX_VALUE.
It's because input parameter for round method is evaluated to float and therefore it returns integer instead of long.